src/KeyFrame.cc

```diff
@@ -622,17 +614,12 @@ void KeyFrame::SetBadFlag() {
     }
   }
 
-  for (std::map<KeyFrame *, int>::iterator mit = mConnectedKeyFrameWeights.begin(),
-                                      mend = mConnectedKeyFrameWeights.end();
-       mit != mend; mit++) {
-    mit->first->EraseConnection(this);
-  }
+  for (auto &pair : mConnectedKeyFrameWeights)
+    pair.first->EraseConnection(this);
```

Comment:
This for loop replacement does not work. EraseConnection modifies the list by erasing elements. This needs to be reworked.